Huawei LTE misc improvements

scop opened the pull request:
## Description:

Misc Huawei LTE cleanup-like improvements.

## Checklist:
  - [x] The code change is tested and works locally.
  - [ ] Local tests pass with `tox`. (Fails due to unrelated reasons) **Your PR cannot be merged unless tests pass**
  - [x] There is no commented out code in this PR.
  - [x] I have followed the [development checklist][dev-checklist]

scop committed:
Constant and whitespace cleanups

MartinHjelmare commented on homeassistant/components/huawei_lte/__init__.py:
```diff
@@ -158,16 +165,18 @@ def _setup_lte(hass, lte_config) -> None:
         mode = "hostname"
-    mac = get_mac_address(**{mode: host})
+    func = partial(get_mac_address, **{mode: host})
+    mac = await hass.async_add_job(func)
```
Please use `hass.async_add_executor_job`.

scop replied:
Done

MartinHjelmare commented on homeassistant/components/huawei_lte/__init__.py:
```diff
@@ -131,16 +138,16 @@ def get_data(self, config):
 
 
-def setup(hass, config) -> bool:
+async def async_setup(hass, config) -> bool:
```
Why do we change to the async api?

scop replied:
Because I'll be moving this component towards config entry based setup, and I believe there's no sync option available there. Also I suppose the smaller threadsafe/executor wrapped sections the better in general.

MartinHjelmare replied:
Fewer context switches is more efficient so if the library doesn't support asyncio we recommend to stay with the sync api.

`async_setup_entry` is a coroutine, yes. `setup` can still be kept a regular function if needed.

MartinHjelmare commented on homeassistant/components/huawei_lte/__init__.py:
```diff
@@ -158,16 +165,18 @@ def _setup_lte(hass, lte_config) -> None:
 
-    connection = AuthorizedConnection(url, username=username, password=password)
+    func = partial(AuthorizedConnection, url, username=username, password=password)
+    connection = await hass.async_add_job(func)
```
Please replace all use of `hass.async_add_job` here with either `hass.async_add_executor_job` or `hass.async_create_task` depending on the target function.

scop committed:
Upgrade huawei_lte_api to 1.3.0

https://github.com/Salamek/huawei-lte-api/releases

scop committed:
Hush traceback if device does not support logout

scop commented:
Thanks for the sanity check. I've dropped the async changes completely at least for now, and added huawei_api_upgrade and logout error hushing for improved support of older/modem devices.

MartinHjelmare reviewed (approved):
Looks good!
